### PR TITLE
Add sidebar to Challenge page

### PR DESCRIPTION
**Pull request by jayaike**
I have added the sidebar to the challenge page as described in other pages

### Screenshots

**After Login**

Screenshot from 2020-01-23 20-21-36

**Before Login**

Screenshot from 2020-01-23 20-21-54

GCI Task: https://codein.withgoogle.com/dashboard/task-instances/6659470118092800/

### CODE REVIEW

**Comment by jayaike** [comment]
@Sanji515 @vkartik97 @Ram81 @RishabhJain2018 PTAL :)

**Comment by Sanji515** [comment]
@nsjcorps Can you please fix the build? Also by looking at the screenshot it's not looking same as in the current version. There's too much extra space on the top, bottom, left and right of card. And yeah you forgot the background color of sidebar also.

**Comment by jayaike** [comment]
Sorry about the build @Sanji515 I'm already on it. I used the classes that were in the pages described in the task. You can take a look

**Comment by jayaike** [comment]
I had not updated my branch. Let me take another screenshot

**Comment by jayaike** [comment]
I did not make another sidebar. I merely imported it as a component. @Sanji515 

The background color has a color...the way it was before.

Please try building on your system so you can take a better look from your end.

**Comment by jayaike** [comment]
I have fixed the build btw @Sanji515

**Comment by Suryansh5545** [comment]
@nsjcorp I think these participate,  host, hosted and create challenges tab shouldn't appear when user is not logged in?

**Comment by jayaike** [comment]
I only bypassed the redirect so I could edit the page. That is why it seems that the I am not authenticated @Suryansh5545

**Comment by Suryansh5545** [comment]
Oh got it:+1:

File: TECHNIQUES_TO_PULL_DATA_FROM_EVALAI.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Techniques to pull data from EvalAI within EvalAI-ngx without using 'git clone' in dockerfile
```
Comment (krtkvrm) [inline TECHNIQUES_TO_PULL_DATA_FROM_EVALAI.md]: Why is this required? Please let me know if I am missing something here.

Comment (jayaike) [reply TECHNIQUES_TO_PULL_DATA_FROM_EVALAI.md]: It's not. It was a mistake and it has been removed.byou can see if you check the branch files @vkartik97

Comment (jayaike) [reply TECHNIQUES_TO_PULL_DATA_FROM_EVALAI.md]: Oh! Sorry! Didn't notice it was still there. My sincere apologies @vkartik97

File: src/app/components/challenge/challenge.component.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export class ChallengeComponent implements OnInit {
    */
   constructor(@Inject(DOCUMENT) document: any, private router: Router, private route: ActivatedRoute,
               private apiService: ApiService, private globalService: GlobalService,
-              private challengeService: ChallengeService, private authService: AuthService,
```
Comment (pushkalkatara) [inline src/app/components/challenge/challenge.component.ts:87]: Do we need this change?

Comment (jayaike) [reply src/app/components/challenge/challenge.component.ts:87]: Yes. It must be public because of the HTML template if conditions. @pushkalkatara

Comment (jayaike) [reply src/app/components/challenge/challenge.component.ts:87]: In the HTML file the footer varies on wether the user is authenticated or not @pushkalkatara

**Comment by Sanji515** [comment]
> @nsjcorps Can you please fix the build? Also by looking at the screenshot it's not looking same as in the current version. There's too much extra space on the top, bottom, left and right of card. And yeah you forgot the background color of sidebar also.

I've checked on my local @nsjcorps , it still doesn't look same as in the current version (the width of challenge container) and please change the background color of sidebar as in the current version.

**Comment by jayaike** [comment]
@Sanji515 PTAL

Screenshot (134)
Screenshot (133)

I also changed it to suit other files.

**Comment by Sanji515** [comment]
Screenshot from 2020-01-19 22-52-02

@nsjcorps Can you please fix this(^) ? This is happening on the challenge page only.

**Comment by jayaike** [comment]
@Sanji515  It looks good on mine as shown in the screenshot. Are you sure you have cloned the updated version?

**Comment by Sanji515** [comment]
> @Sanji515 It looks good on mine as shown in the screenshot. Are you sure you have cloned the updated version?

Yes, I'm on the updated version, try when user is logged in.

**Comment by jayaike** [comment]
Is it only on the challenge detail page? or challenge list

**Comment by Sanji515** [comment]
Also can we be consistent by removing the sidebar

> Is it only on the challenge detail page? or challenge list

Challenge detail page.

**Comment by jayaike** [comment]
I don't understand

**Comment by Sanji515** [comment]
Screenshot from 2020-01-19 22-52-02


It is happening on the challenge overview page when user is logged in.

**Comment by jayaike** [comment]
I updated it already

**Comment by jayaike** [comment]
Unfortunately, I am still not seeing that error on challenge page

File: src/app/components/challenge/challenge.component.html [hunk 3]
```diff
@@ -147,7 +147,7 @@
                 </div>
             </div>
         </div>
-        <app-footer [isDash]="true" *ngIf="authService.isAuth"></app-footer>
+        <app-footer [isDash]="true" *ngIf="!authService.isAuth"></app-footer>
     </div>
 </div>
-<app-footer *ngIf="!authService.isAuth"></app-footer>
```
Comment (Sanji515) [inline src/app/components/challenge/challenge.component.html]: @nsjcorps It is happening due to this change, please revert it.

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html]: Thank you so much @Sanji515 . I used it to bypass the view so I could edit it. It's really  hard doing all these tasks without docker

**Comment by jayaike** [comment]
How does it look @Sanji515 ?

**Comment by Sanji515** [comment]
Lets keep it consistent with the current version, so please remove the sidebar if user is not logged in

**Comment by jayaike** [comment]
@Sanji515 done

**Comment by jayaike** [comment]
@Sanji515 PTAL

**Comment by Sanji515** [comment]
There's too much extra space on the top, bottom, left and right of card for the resolution between 990px and 1130px. Can you please fix it?

**Comment by jayaike** [comment]
Do you mean margin or padding? @Sanji515

**Comment by jayaike** [comment]
I just reduced the margin from the sides @Sanji515 PTAL

**Comment by jayaike** [comment]
Sorry the push took some time @Sanji515

**Comment by Sanji515** [comment]
> I just reduced the margin from the sides @Sanji515 PTAL

Or just remove this from [here](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/challenge/challenge.component.scss#L19) for the small resolutions.

**Comment by jayaike** [comment]
Done @Sanji515

File: src/app/components/challenge/challenge.component.html [hunk 4]
```diff
@@ -1,150 +1,153 @@
...
-                                        {{challenge['creator']['team_name']}}
-                                    </a>
-                                </span>
-                                <span style="line-height: 1.5"
-                                    *ngIf="challenge.creator.team_url == ''"><strong
-                                        class="text-light-black fw-semibold
-                                        fs-15">Organized by:</strong>
-                                    {{challenge['creator']['team_name']}}
-                                </span>
+<div class="web-container" [style.center]="authService.isAuth">
```
Comment (Sanji515) [inline src/app/components/challenge/challenge.component.html]: You need to center the container when user is not logged in. Currently the UI is right aligned on the challenge overview page when the user is not loggeed in. @nsjcorps Please look into it.

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html]: Done @Sanji515

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html]: PTAL @Sanji515

Comment (Sanji515) [reply src/app/components/challenge/challenge.component.html]: Still the same @nsjcorps

**Comment by jayaike** [comment]
Sorry @Sanji515 I edited the wrong div

**Comment by jayaike** [comment]
Thanks for bearing with me though

**Comment by jayaike** [comment]
I also had to make the changes to the other files @Sanji515

**Comment by Sanji515** [comment]
Hey, still the same. Always please try checking on your local before pushing.

**Comment by jayaike** [comment]
@Sanji515 I tested it this time. It is centered

**Comment by jayaike** [comment]
localhost_4200_challenge_2_overview - Google Chrome 1_19_2020 9_42_55 PM

**Comment by jayaike** [comment]
@Sanji515 PTAL

File: src/app/components/dashboard/dashboard.component.html [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <app-side-bar *ngIf="authService.isAuth"></app-side-bar>
 <app-header-static></app-header-static>
-<div class="web-container" [style.center]="!authService.isAuth">
-  <div class="dashboard-flex">
+<div class="web-container">
+  <div class="dashboard-flex" [class.center]="!authService.isAuth">
```
Comment (Sanji515) [inline src/app/components/dashboard/dashboard.component.html]: Do we really need center class here and at similar other files cause this page is only available when user is logged in?

**Comment by jayaike** [comment]
That is the only way the `&.center` css gets called

**Comment by nsjcorps** [comment]
That is the only way the `&.center` css gets called

**Comment by Sanji515** [comment]
> That is the only way the `&.center` css gets called

Yes, but the condition gets only applied when the user is not logged in right? Then what is the use of it on the dashboard page and similar other.

**Comment by jayaike** [comment]
Yes that is true. What do you suggest @Sanji515

**Comment by jayaike** [comment]
Because it should be tested

**Comment by Sanji515** [comment]
Lets remove all those `class.center` class related changes from the files like analytics, dashboard and similar other.

**Comment by jayaike** [comment]
How will I center the div then @Sanji515 ?

**Comment by Sanji515** [comment]
> How will I center the div then @Sanji515

I'm not saying to remove from the `challenge.component.html` as it is gonna be used as we displaying the page even when user is not authenticated but the pages like `dashboard` and similar others are not gonna shown if the user is not authenticated so what's the use of `class.center="!authService.isAuth"`. So lets remove the code from these files.

**Comment by jayaike** [comment]
Okay I get now. They are protected routes so unauthorized users wont see them anyway.

So should I remove them now? @Sanji515

**Comment by Sanji515** [comment]
> Okay I get now. They are protected routes so unauthorized users wont see them anyway.
> 
> So should I remove them now? @Sanji515

yes go ahead!

**Comment by jayaike** [comment]
Should I also remove from public lists?

**Comment by Sanji515** [comment]
> Should I also remove from public lists?

nope, not from here.

**Comment by jayaike** [comment]
Alright. I removed the one's for the protected routes

**Comment by Sanji515** [comment]
Also please remove its css if its not using.

**Comment by jayaike** [comment]
Done! @Sanji515

File: src/app/components/challenge/challenge.component.html [hunk 6]
```diff
@@ -1,150 +1,153 @@
...
-                                            <i class="{{publishChallenge.icon}}"
-                                                aria-hidden="true"></i>
-                                        </a>
-                                    </span>
-                                </span>
-                                <br>
-                                <mat-checkbox color="primary" class="fw-semibold" *ngIf="isChallengeHost"
-                                    [checked]="!challenge['is_registration_open']"
-                                    (click)="stopParticipation($event)">Close Participation
-                                </mat-checkbox>
```
Comment (Sanji515) [inline src/app/components/challenge/challenge.component.html:71]: Hey, why have you removed `stop participation` option?

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html:71]: I don't think I did.

**Comment by jayaike** [comment]
Let me revert

**Comment by jayaike** [comment]
Reverted! I guess it happened when I was trying to resolve conflicts. Thanks for showing me!

File: src/app/components/challenge/challenge.component.html [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <app-side-bar *ngIf="authService.isAuth"></app-side-bar>
 <app-header-static></app-header-static>
-<div class="web-container">
-    <div class="challenge-flex" [class.center]="!authService.isAuth">
+<div class="web-container" [style.center]="authService.isAuth">
+    <div class="challenge-flex">
```
Comment (Sanji515) [inline src/app/components/challenge/challenge.component.html]: Why have you removed this from here now?

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html]: Sorry about that. I am a little helter-skelter...Reverted!

Comment (Sanji515) [reply src/app/components/challenge/challenge.component.html]: You need to add that this thing with `challenge-flex` in order to work.

Note: Please always check if its working on your local before pushing.

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html]: I reverted the `!` too

Comment (Sanji515) [reply src/app/components/challenge/challenge.component.html]: It will not work, you need to add that with the `challenge-flex` div.

> Note: Please always check if its working on your local before pushing.

. :smile:

File: src/app/components/challenge/challenge.component.html [hunk 8]
```diff
@@ -1,150 +1,158 @@
...
-                                    </a>
-                                </span>
-                                <span style="line-height: 1.5"
-                                    *ngIf="challenge.creator.team_url == ''"><strong
-                                        class="text-light-black fw-semibold
-                                        fs-15">Organized by:</strong>
-                                    {{challenge['creator']['team_name']}}
-                                </span>
+<div class="web-container" [class.center]="!authService.isAuth">
+    <div class="challenge-flex">
```
Comment (Sanji515) [inline src/app/components/challenge/challenge.component.html]: @nsjcorps  You need to add `[class.center]="!authService.isAuth"` on this(^) `div`.

Comment (jayaike) [reply src/app/components/challenge/challenge.component.html]: alright...done!

File: src/app/components/publiclists/publiclists.component.html [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <app-side-bar *ngIf="authService.isAuth"></app-side-bar>
 <app-header-static></app-header-static>
-<div class="web-container" [style.center]="!authService.isAuth">
-  <div class="dashboard-flex">
+<div class="web-container">
+  <div class="dashboard-flex" [class.center]="!authService.isAuth">
```
Comment (Sanji515) [inline src/app/components/publiclists/publiclists.component.html]: Please move this condition to `web-container` div.

File: src/app/components/publiclists/publiclists.component.scss [hunk 10]
```diff
@@ -12,20 +20,13 @@
 }
 
 .web-container {
-  width: calc(100vw - 223px);
-  float: right;
+  background-color: #fafafa;
```
Comment (Sanji515) [inline src/app/components/publiclists/publiclists.component.scss]: Please revert all the changes of this file leaving this `background-color` change.

File: src/app/components/dashboard/dashboard.component.scss [hunk 11]
```diff
@@ -12,22 +14,15 @@
 }
 
 .web-container {
-  width: calc(100vw - 223px);
-  float: right;
```
Comment (Sanji515) [inline src/app/components/dashboard/dashboard.component.scss]: Please revert this change (keeping `background-color`)

File: src/app/components/challenge-create/challenge-create.component.scss [hunk 12]
```diff
@@ -12,16 +14,9 @@
 }
 
 .web-container {
-  width: calc(100vw - 223px);
-  float: right;
```
Comment (Sanji515) [inline src/app/components/challenge-create/challenge-create.component.scss]: Same here.

File: src/app/components/analytics/analytics.component.scss [hunk 13]
```diff
@@ -12,22 +14,15 @@
 }
 
 .web-container {
-  width: calc(100vw - 223px);
-  float: right;
```
Comment (Sanji515) [inline src/app/components/analytics/analytics.component.scss]: Same here.

File: src/app/components/publiclists/publiclists.component.html [hunk 14]
```diff
@@ -1,6 +1,6 @@
 <app-side-bar *ngIf="authService.isAuth"></app-side-bar>
 <app-header-static></app-header-static>
-<div class="web-container" [style.center]="!authService.isAuth">
+<div class="web-container"  [class.center]="!authService.isAuth">
```
Comment (Sanji515) [inline src/app/components/publiclists/publiclists.component.html]: Please remove the extra space from here.

File: src/app/components/analytics/analytics.component.scss [hunk 15]
```diff
@@ -24,8 +25,6 @@
   }
 }
 
-/*media queries*/
-
```
Comment (Sanji515) [inline src/app/components/analytics/analytics.component.scss]: Please revert this change.

File: src/app/components/dashboard/dashboard.component.scss [hunk 16]
```diff
@@ -24,8 +25,6 @@
   }
 }
 
-/*media queries*/
-
```
Comment (Sanji515) [inline src/app/components/dashboard/dashboard.component.scss]: Same here.

File: src/app/components/analytics/analytics.component.scss [hunk 17]
```diff
@@ -16,14 +16,14 @@
...
   overflow-x: hidden;
+  background-color: #fafafa;
   &.center {
     float: none;
     margin: 0 auto;
     text-align: left;
     overflow: hidden;
   }
 }
-
```
Comment (Sanji515) [inline src/app/components/analytics/analytics.component.scss]: Please revert this change.

File: src/app/components/dashboard/dashboard.component.scss [hunk 18]
```diff
@@ -16,14 +16,14 @@
...
   overflow-x: hidden;
+  background-color: #fafafa;
   &.center {
     float: none;
     margin: 0 auto;
     text-align: left;
     overflow: hidden;
   }
 }
-
```
Comment (Sanji515) [inline src/app/components/dashboard/dashboard.component.scss]: Please revert this change.

File: src/app/components/challenge-create/challenge-create.component.scss [hunk 19]
```diff
@@ -65,7 +66,7 @@
 /*media queries*/
 
 @media only screen and (max-width: $med-screen) {
-  .web-container {
+  .challenge-create-flex {
```
Comment (Sanji515) [inline src/app/components/challenge-create/challenge-create.component.scss]: Please revert this change too. (it should set the width to 100% of web-container div)

Comment (jayaike) [reply src/app/components/challenge-create/challenge-create.component.scss]: Done @Sanji515

**Comment by Sanji515** [comment]
@nsjcorps You need to change the background color of sidebar as in the current version.

**Comment by jayaike** [comment]
That means I would also have to change the height because the sidebar does not occupy 100% height currently and if I change the background color it will look weird. Should I adjust the height too?

**Comment by jayaike** [comment]
Oh I see...there is no need to adjust the height because the container already has the background color

**Comment by jayaike** [comment]
Done @Sanji515

**Comment by Sanji515** [comment]
Screenshot from 2020-01-20 05-33-32

But you need to fix this(^) thing.

**Hint:** There is some `top` value given to the outer div. You need to remove that from the outer div and have to add some top value or margin to inner div.

**Comment by jayaike** [comment]
@Sanji515 What do you think of the changes?

**Comment by Sanji515** [comment]
> @Sanji515 What do you think of the changes?

Now the scroll disappears as the links have the sufficient height and it is not consistent with the current version, in the current version sidebar links starts with some top value due to which scroll shows up.

Can you please make this same as in the current version?

**Comment by jayaike** [comment]
I will increase the padding to 128px

**Comment by Sanji515** [comment]
> I will increase the padding to 128px

yes, sounds good!

**Comment by jayaike** [comment]
Done! @Sanji515

**Comment by Sanji515** [comment]
But you also need to fix that `border-right`, it will show up as we have removed the `top` value.

**Comment by jayaike** [comment]
I have removed it @Sanji515 . How does it look now?

File: src/app/components/utility/side-bar/side-bar.component.scss [hunk 20]
```diff
@@ -1,13 +1,13 @@
 @import "styles/variables";
 .ev-sidebar {
   display: block;
-  height: calc(100vh - 120px);
+  height: 100vh;
   width: 220px;
+  padding-top: 128px;
```
Comment (Sanji515) [inline src/app/components/utility/side-bar/side-bar.component.scss]: Can you please change this to 98px?

Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: I have now

**Comment by Sanji515** [comment]
> I have removed it @Sanji515 . How does it look now?

But we cannot remove it completely, it does not have effect on small devices as it is hidden under the scroll but we cannot ignore devices with higher resolution.

**Comment by jayaike** [comment]
Should I add a media query for it to show on medium and large devices?

**Comment by Sanji515** [comment]
> Should I add a media query for it to show on medium and large devices?

Its related to height not width, if some device height is large enough then scroll will not be shown there we need border on the right.

**Comment by Sanji515** [comment]
> Hint: There is some top value given to the outer div. You need to remove that from the outer div and have to add some top value or margin to inner div.

Instead you can do this(^) way, you can apply `border-right` and margin as equal to the top value to the inner div with the class name `links-section-outer`

**Comment by jayaike** [comment]
I don't understand @Sanji515

**Comment by jayaike** [comment]
How does this look @Sanji515

**Comment by Sanji515** [comment]
Screenshot from 2020-01-20 15-59-30

I can see that border-right on the top which doesn't look good, please look to it @nsjcorps

**Comment by jayaike** [comment]
I did some math by removing the height of the navabr from the sidebar then setting it's top value to the height of the navabr. I then reduced the padding to 98px minus height of navbar.

Sorry I am using my phone. My laptop crashed as a result of the load so I had to reboot it

IMG_20200120_123723_890

**Comment by jayaike** [comment]
@Sanji515 PTAL

**Comment by Sanji515** [comment]
Hmm... this way the scroll will start after the margin of `34px` and also the margin-right is still showing above the scroll. Just moving the `border-right` to class `links-section-outer` will fix the margin-right thing but it will not be consistent as in the current version.

To make it consistent witht the current version, remove the top value from the `ev-sidebar` class and move the `border-right` to `links-section-outer` class and also add margin-top of `98px` to this class and rest revert all the changes that you have did.

Let me know if you didn't get this @nsjcorps

**Comment by jayaike** [comment]
@Sanji515 Ok but what do I do with the height of the ev sidebar

**Comment by Sanji515** [comment]
> @Sanji515 Ok but what do I do with the height of the ev sidebar

Let it be the same as before.

**Comment by jayaike** [comment]
That is I should leave it as it is @Sanji515

**Comment by jayaike** [comment]
Or I should revert all the changes the original way it was after adding margintop 98px to links-section-outer and removing top value from ev sidebar

**Comment by Sanji515** [comment]
> Or I should revert all the changes the original way it was after adding margintop 98px to links-section-outer and removing top value from ev sidebar

Yes, also move the border-right from `ev-sidebar` to `links-section-outer`.

**Comment by jayaike** [comment]
That means I also need to add background color to the link section outer because currently it is transparent toward the botton

**Comment by jayaike** [comment]
I did that and the sidebar does not cover the full height anymore

**Comment by Sanji515** [comment]
> That means I also need to add background color to the link section outer because currently it is transparent toward the botton

no, add the background-color to `ev-sidebar`.

**Comment by jayaike** [comment]
Ok but when I reverted the changes the sidebar did not occupy full height again

**Comment by Sanji515** [comment]
> Ok but when I reverted the changes the sidebar did not occupy full height again

Please push the changes that you have done.

**Comment by jayaike** [comment]
@Sanji515  Alright....I have pushed it

**Comment by Sanji515** [comment]
It looks good on mine, and it is same as in the current version. Can you please paste screenshot how it looks on yours @nsjcorps

**Comment by jayaike** [comment]
Maybe mine is happening because of all the tweaks and hacks I have on my version of the project @Sanji515

**Comment by jayaike** [comment]
But I doubt though. Should I try and make some changes let's see if it works for both of us 😁

**Comment by jayaike** [comment]
This change I made looks like this @Sanji515 



Can you test on your system?

**Comment by jayaike** [comment]
The sidebar border right doesn't go past anymore and it shows well over here

**Comment by Sanji515** [comment]
> The sidebar border right doesn't go past anymore and it shows well over here

@nsjcorps Yes, this looks good, but I think it is not same as in the current version. [Here](https://evalai.cloudcv.org/web/dashboard) the scroll doesn't touches the bottom.

**Comment by jayaike** [comment]
I just added a padding that is equal to that of the top so it is equal just like the current version of EvalAI @Sanji515

File: src/app/components/utility/side-bar/side-bar.component.scss [hunk 21]
```diff
@@ -7,6 +7,7 @@
   color: #fff;
   z-index: 10;
   text-align: center;
+  padding-bottom: 34px;
```
Comment (Sanji515) [inline src/app/components/utility/side-bar/side-bar.component.scss]: Instead I would suggest to revert to [this](https://github.com/Cloud-CV/EvalAI-ngx/pull/277/commits/b73cafad8c50ed0cb0c566424d73b94ec6fb6141) commit.

**Comment by jayaike** [comment]
Ok @Sanji515 . Done. I had to add that 34 px to the 98px already there

**Comment by jayaike** [comment]
Do you understand @Sanji515

**Comment by jayaike** [comment]
This is the logic here.

**Comment by Sanji515** [comment]
| Latest Commit        | Current version (EvalAI)  |
| ------------- |:-------------:|
| Screenshot from 2020-01-20 23-31-39 |Screenshot from 2020-01-20 23-31-54 |

Please look at the difference, Can't see all the links in one view @nsjcorps

File: src/app/components/utility/side-bar/side-bar.component.scss [hunk 22]
```diff
@@ -16,8 +14,10 @@
 }
 
 .links-section-outer {
+  border-right: 1px solid $grad-light-black;
   overflow: auto;
-  height: 100%;
+  height: calc(100% - 132px);
```
Comment (Sanji515) [inline src/app/components/utility/side-bar/side-bar.component.scss]: You should change it to `calc(100% - 120px);`

**Comment by jayaike** [comment]
~`calc(100% - 104px)` makes it the same as current version...PTAL @Sanji515~

File: src/app/components/utility/side-bar/side-bar.component.scss (hunk 22 above)
Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: > You should change it to `calc(100% - 120px);`

O ok..I'll do it now

Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: Done @Sanji515

File: src/app/components/utility/side-bar/side-bar.component.scss [hunk 23]
```diff
@@ -16,8 +14,10 @@
 }
 
 .links-section-outer {
+  border-right: 1px solid $grad-light-black;
   overflow: auto;
-  height: 100%;
+  height: calc(100% - 120px);
+  margin-top: 92px;
```
Comment (Sanji515) [inline src/app/components/utility/side-bar/side-bar.component.scss]: Please change it to 98px

Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: done!! @Sanji515

**Review by Sanji515 — COMMENTED** [review]
Screenshot from 2020-01-21 00-38-05

Can you please fix this (^)? (When user is not logged in)

**Comment by jayaike** [comment]
What am I to fix? @Sanji515

**Comment by Sanji515** [comment]
> What am I to fix? @Sanji515

Background color is not applied to whole mid-container instead it has some margin to the left and right.

**Comment by jayaike** [comment]
What is it supposed to look like @Sanji515

**Comment by Sanji515** [comment]
> What is it supposed to look like @Sanji515

Same as in the current version ([here](https://evalai.cloudcv.org/web/challenges/list))

**Comment by jayaike** [comment]
Just did it @Sanji515

**Comment by Sanji515** [comment]
> Just did it @Sanji515

Hey it doesn't look same (challenge card width).

I would suggest you to add `background-color: #fafafa` to the body tag in `styles/base.scss` and remove the background-color from all the parent component. @nsjcorps what do you say? Try checking this change if this works.

**Comment by jayaike** [comment]
As for the background color @Sanji515 . I would say that would be rather complex for this task because other changes will be involved

**Comment by jayaike** [comment]
The GCI task was more about setting the sidebar in the challenge page

**Comment by Sanji515** [comment]
> As for the background color @Sanji515 . I would say that would be rather complex for this task because other changes will be involved

yeah but with this change it doesn't look good. At least you can give a try (just add the background in the body in `base.scss` and remove from all other parent component). Lets just see if this will work or not.

**Comment by jayaike** [comment]
Sorry @Sanji515 I was supposed to divide 223px by 2. Which I have done now.

**Comment by jayaike** [comment]
Change has been made @Sanji515 

> > As for the background color @Sanji515 . I would say that would be rather complex for this task because other changes will be involved
> 
> yeah but with this change it doesn't look good. At least you can give a try (just add the background in the body in `base.scss` and remove from all other parent component). Lets just see if this will work or not.

**Comment by Sanji515** [comment]
> Change has been made @Sanji515

Have you tested on your local?

**Comment by jayaike** [comment]
I tested but I can't test all routes because of my docker issue and my laptop issue but as for the challenge one it is good

File: src/app/components/challenge-create/challenge-create.component.scss [hunk 24]
```diff
@@ -12,10 +12,10 @@
 }
 
 .web-container {
-  width: calc(100vw - 223px);
-  float: right;
   padding-top: 70px;
   overflow-x: hidden;
+  width: calc(100vw - 223px);
+  float: right;
```
Comment (Sanji515) [inline src/app/components/challenge-create/challenge-create.component.scss]: Please revert all the changes of this file.

File: src/app/components/publiclists/publiclists.component.scss [hunk 25]
```diff
@@ -21,6 +21,9 @@
     margin: 0 auto;
     text-align: left;
     overflow: hidden;
+    width: 100%;
+    padding-left: 111.5px;
+    padding-right: 111.5px;
```
Comment (Sanji515) [inline src/app/components/publiclists/publiclists.component.scss]: Please revert this change too.

File: src/app/components/utility/side-bar/side-bar.component.scss [hunk 26]
```diff
@@ -1,13 +1,11 @@
 @import "styles/variables";
 .ev-sidebar {
   display: block;
-  height: calc(100vh - 120px);
+  height: 100vh;
   width: 220px;
-  border-right: 1px solid $grad-light-black;
-  background-color: $transparent;
+  background-color: #fafafa;
```
Comment (Sanji515) [inline src/app/components/utility/side-bar/side-bar.component.scss]: Now, we don't this change here.

Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: we still need it because when we scroll to the bottom the sidebar will be messed up

Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: because of the footer

Comment (Sanji515) [reply src/app/components/utility/side-bar/side-bar.component.scss]: but why?

File: src/app/components/challenge-create/challenge-create.component.scss (hunk 24 above)
Comment (jayaike) [reply src/app/components/challenge-create/challenge-create.component.scss]: Then how will the challenge-flex divs be styled?

File: src/app/components/publiclists/publiclists.component.scss (hunk 25 above)
Comment (jayaike) [reply src/app/components/publiclists/publiclists.component.scss]: Done!

File: src/app/components/challenge-create/challenge-create.component.scss (hunk 24 above)
Comment (Sanji515) [reply src/app/components/challenge-create/challenge-create.component.scss]: No actually these changes were already there na on line no 15 and 16, so let it be there only.

File: src/app/components/utility/side-bar/side-bar.component.scss (hunk 26 above)
Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: because the side bar is fixed

File: src/app/components/challenge-create/challenge-create.component.scss (hunk 24 above)
Comment (jayaike) [reply src/app/components/challenge-create/challenge-create.component.scss]: I'm confused here on what you want me to do

File: src/app/components/utility/side-bar/side-bar.component.scss (hunk 26 above)
Comment (jayaike) [reply src/app/components/utility/side-bar/side-bar.component.scss]: if the sidebar doesnt have that background color when we scroll over the footer it will not show properly. You cann test it and see

**Comment by jayaike** [comment]
We seem to be going in circles @Sanji515

File: src/app/components/challenge-create/challenge-create.component.scss (hunk 24 above)
Comment (Sanji515) [reply src/app/components/challenge-create/challenge-create.component.scss]: Just make this file as we have in the master branch of this repo so that this file doesn't contain any change we already have the same width and float value.

File: src/app/components/utility/side-bar/side-bar.component.scss (hunk 26 above)
Comment (Sanji515) [reply src/app/components/utility/side-bar/side-bar.component.scss]: Yes, I've tested, it's working on my local.

**Comment by jayaike** [comment]
@Sanji515 :)....I did the background change and i reverted the other file

**Comment by Sanji515** [comment]
> We seem to be going in circles @Sanji515

I think we came a long way @nsjcorps :smile:

**Review by Sanji515 — APPROVED** [review]
Thanks @nsjcorps you did a great job :+1: LGTM! This can be merged now :tada:

**Comment by jayaike** [comment]
> > We seem to be going in circles @Sanji515
> 
> I think we came a long way @nsjcorps 😄

Oh yes we did!

**Comment by jayaike** [comment]
> Thanks @nsjcorps you did a great job 👍 LGTM! This can be merged now 🎉

Thank you....63 commits and 171 + 1 comments 😂😂

I have submitted the GCI task for review

**Comment by jayaike** [comment]
I think I just did the 100th task. That has got to be a Landmark 🎉🎉🎉🎉🎉

**Comment by Sanji515** [comment]
@nsjcorps Can you please resolve conflicts and update the PR description here?

**Comment by jayaike** [comment]
What should I update the PR description to? @Sanji515 Sorry, I am just seeing this.

**Comment by Sanji515** [comment]
> What should I update the PR description to? @Sanji515 Sorry, I am just seeing this.

Current screenshot showing the sidebar when user is not logged in but in our latest commit the sidebar should come after login and not before login.

**Comment by jayaike** [comment]
Hi @Sanji515 , unfortunately I cannot log in to the evalai-ngx Dev environment. Can you take the screenshots on your end and give them to me so I can update them. It would be highly appreciated 😁😁

**Comment by Sanji515** [comment]
> Hi @Sanji515 , unfortunately I cannot log in to the evalai-ngx Dev environment. Can you take the screenshots on your end and give them to me so I can update them. It would be highly appreciated

No worries @nsjcorps , I've updated your PR description with the screenshots.

**Comment by jayaike** [comment]
Thanks @Sanji515
